File: cpp/unittests/SearchInRotatedSortedArrayTest.cpp
#include "SearchInRotatedSortedArray.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(SearchInRotatedSortedArray, Solution33_1) {
  std::vector<int> Nums{4, 5, 6, 7, 0, 1, 2};
  const int Target1 = 0, Expected1 = 4;
  EXPECT_EQ(Expected1, Solution33_1().search(Nums, Target1));

  const int Target2 = 3, Expected2 = -1;
  EXPECT_EQ(Expected2, Solution33_1().search(Nums, Target2));
}

File: java/src/test/java/io/binac/leetcode/SearchInRotatedSortedArrayTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class SearchInRotatedSortedArrayTest {
    private static final SearchInRotatedSortedArray.Solution1 SOLUTION1 = new SearchInRotatedSortedArray.Solution1();

    private final int[] NUMS1 = {4, 5, 6, 7, 0, 1, 2};
    private final int TARGET1 = 0;
    private final int EXPECTED1 = 4;

    private final int[] NUMS2 = {4, 5, 6, 7, 0, 1, 2};
    private final int TARGET2 = 3;
    private final int EXPECTED2 = -1;

    private final int[] NUMS3 = {1};
    private final int TARGET3 = 0;
    private final int EXPECTED3 = -1;

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.search(NUMS1, TARGET1));
        assertEquals(EXPECTED2, SOLUTION1.search(NUMS2, TARGET2));
        assertEquals(EXPECTED3, SOLUTION1.search(NUMS3, TARGET3));
    }
}